ql/src/java/org/apache/hadoop/hive/ql/ddl/table/column/drop/AlterTableDropColumnOperation.java

```diff
@@ -0,0 +1,95 @@
+import org.apache.hadoop.hive.metastore.api.FieldSchema;
+import org.apache.hadoop.hive.metastore.api.StorageDescriptor;
+import org.apache.hadoop.hive.ql.ErrorMsg;
+import org.apache.hadoop.hive.ql.ddl.DDLOperationContext;
+import org.apache.hadoop.hive.ql.ddl.table.AbstractAlterTableOperation;
+import org.apache.hadoop.hive.ql.ddl.table.AlterTableUtils;
+import org.apache.hadoop.hive.ql.io.orc.OrcSerde;
+import org.apache.hadoop.hive.ql.io.parquet.serde.ParquetHiveSerDe;
+import org.apache.hadoop.hive.ql.metadata.HiveException;
+import org.apache.hadoop.hive.ql.metadata.Partition;
+import org.apache.hadoop.hive.ql.metadata.Table;
+import org.apache.hadoop.hive.serde2.MetadataTypedColumnsetSerDe;
+import org.apache.hadoop.hive.serde2.avro.AvroSerdeUtils;
+import org.apache.hadoop.hive.serde2.columnar.ColumnarSerDe;
+import org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Operation process of dropping column.
+ */
+public class AlterTableDropColumnOperation extends AbstractAlterTableOperation<AlterTableDropColumnDesc> {
+
+    public AlterTableDropColumnOperation(DDLOperationContext context, AlterTableDropColumnDesc desc) {
+        super(context, desc);
+    }
+
+    private static final Set<String> VALID_SERIALIZATION_LIBS = ImmutableSet.of(
+            MetadataTypedColumnsetSerDe.class.getName(), LazySimpleSerDe.class.getName(), ColumnarSerDe.class.getName(),
+            ParquetHiveSerDe.class.getName(), OrcSerde.class.getName(), "org.apache.iceberg.mr.hive.HiveIcebergSerDe");
+
+    @Override
+    protected void doAlteration(Table table, Partition partition) throws HiveException {
+        StorageDescriptor sd = getStorageDescriptor(table, partition);
+        String serializationLib = sd.getSerdeInfo().getSerializationLib();
+        AvroSerdeUtils.handleAlterTableForAvro(context.getConf(), serializationLib, table.getTTable().getParameters());
+
+        if ("org.apache.hadoop.hive.serde.thrift.columnsetSerDe".equals(serializationLib)) {
+            context.getConsole().printInfo("Dropping column for columnsetSerDe and changing to LazySimpleSerDe");
+            sd.getSerdeInfo().setSerializationLib(LazySimpleSerDe.class.getName());
+        }
+
+        boolean isOrcSchemaEvolution = serializationLib.equals(OrcSerde.class.getName()) &&
+                AlterTableUtils.isSchemaEvolutionEnabled(table, context.getConf());
+        if (isOrcSchemaEvolution) {
+            throw new HiveException(ErrorMsg.CANNOT_DROP_COLUMN, desc.getDbTableName());
+        }
+
+        if (ParquetHiveSerDe.isParquetTable(table) && AlterTableUtils.isSchemaEvolutionEnabled(table, context.getConf()) &&
+                !desc.isCascade() && table.isPartitioned()) {
+            LOG.warn("Cannot drop column from a partitioned parquet table without the CASCADE option");
+            throw new HiveException(ErrorMsg.CANNOT_DROP_COLUMN, desc.getDbTableName());
+        }
```

Comment:
nit: Do we need this `warn` if we are throwing an exception too? Leaving the `warn` is fine too, so I will leave it up to you.